### PR TITLE
Bugfix missing term

### PR DESCRIPTION
**Pull request by cgutteridge**

### CODE REVIEW

File: test/filters/FreeTextFilter.test.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import { FreeTextFilter } from '../../src/filters/FreeTextFilter'
+import { describe, expect, test } from 'vitest'
+
+describe('FreeTextFilter', () => {
+  describe('matchesRecord()', () => {
```
Comment (dp11g14) [inline test/filters/FreeTextFilter.test.js:5]: Still slightly disagree with your test name structure as I don't feel it reads easily. My preference would be FreeTextFilter matchesRecord _returns true when the record matches a term in the state._

We were supposed to discuss this with the team but didn't really have another standup after that. We could maybe discuss/vote on teams?

File: test/filters/FreeTextFilter.test.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    })
+
+    test('when the record matches an integer value: return true', () => {
+      const record = { title: { field: {}, value: 17 } }
+      const freeTextFilter = new FreeTextFilter({ term: '17' })
+
+      expect(freeTextFilter.matchesRecord(record)).toEqual(true)
+    })
+
+    test('when the record does not contain the search terms spread over several fields: return false', () => {
```
Comment (dp11g14) [inline test/filters/FreeTextFilter.test.js:34]: Not sure what _spread over several fields_ means here - its quite vague language. Would _when all the search terms do not appear anywhere in the record's fields_ describe what you are testing?

File: src/filters/FreeTextFilter.js [hunk 3]
```diff
@@ -11,6 +11,12 @@ export class FreeTextFilter extends Filter {
       let termFound = false
       const fieldNames = Object.keys(record)
       fieldNames.forEach((fieldName) => {
+        if (typeof record[fieldName] !== 'object') { return }
```
Comment (dp11g14) [inline src/filters/FreeTextFilter.js:14]: Do all these checks have equivalent tests?

Comment (cgutteridge) [reply src/filters/FreeTextFilter.js:14]: not quite; the final test should fail if any of them fail. Hopefully this code will be rethought next version anyway.

File: test/filters/FreeTextFilter.test.js (hunk 2 above)
Comment (cgutteridge) [reply test/filters/FreeTextFilter.test.js:34]: I'll reword.

**Pull request merged**